app/js/addCategories.ts

```diff
@@ -0,0 +1,45 @@
+async function addCategory ():Promise<void> {
+    let categoriesObject = await getData('/category');
+    document.querySelector<HTMLFormElement>('.categoriesForm').addEventListener('submit', (e) => {
+        e.preventDefault();
+        let newCategory: string = document.querySelector<HTMLFormElement>('#categoryName').value;
+        let responseMessage = document.querySelector('#addedCategoryConfirmation');
+        let listOfCategories = [];
+        categoriesObject.data.forEach((category) => {
+            listOfCategories.push(category.name);
+        })
+
+        if (!listOfCategories.includes(newCategory)) {
```

Comment:
technically not a story requirement. did you check with the PO if they want this?

Reply:
No, but surely you do?

Reply:
yes, but never assume that. always ask these things